packages/mobile/src/exchange/ExchangeReview.tsx

```diff
@@ -1,133 +1,229 @@
-import ReviewFrame from '@celo/react-components/components/ReviewFrame'
-import ReviewHeader from '@celo/react-components/components/ReviewHeader'
+import Button, { BtnTypes } from '@celo/react-components/components/Button'
 import colors from '@celo/react-components/styles/colors'
+import { fontStyles } from '@celo/react-components/styles/fonts'
+import { componentStyles } from '@celo/react-components/styles/styles'
 import BigNumber from 'bignumber.js'
 import * as React from 'react'
 import { withNamespaces, WithNamespaces } from 'react-i18next'
-import { StyleSheet } from 'react-native'
+import { ScrollView, StyleSheet, Text, View } from 'react-native'
 import SafeAreaView from 'react-native-safe-area-view'
 import { NavigationInjectedProps } from 'react-navigation'
 import { connect } from 'react-redux'
 import CeloAnalytics from 'src/analytics/CeloAnalytics'
 import { CustomEventNames } from 'src/analytics/constants'
 import componentWithAnalytics from 'src/analytics/wrapper'
 import { exchangeTokens, fetchExchangeRate } from 'src/exchange/actions'
-import ExchangeConfirmationCard from 'src/exchange/ExchangeConfirmationCard'
+import { ExchangeHeader } from 'src/exchange/ExchangeHeader'
+import FeeIcon from 'src/exchange/FeeExchangeIcon'
 import { ExchangeRatePair } from 'src/exchange/reducer'
-import { CURRENCY_ENUM as Token } from 'src/geth/consts'
+import { CURRENCY_ENUM } from 'src/geth/consts'
 import { Namespaces } from 'src/i18n'
-import { navigate, navigateBack } from 'src/navigator/NavigationService'
+import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { RootState } from 'src/redux/reducers'
 import { isAppConnected } from 'src/redux/selectors'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
-import {
-  getNewDollarBalance,
-  getNewGoldBalance,
-  getRateForMakerToken,
-  getTakerAmount,
-} from 'src/utils/currencyExchange'
+import { getRateForMakerToken, getTakerAmount } from 'src/utils/currencyExchange'
 import { getMoneyDisplayValue } from 'src/utils/formatting'
 
 interface StateProps {
-  dollarBalance: string | null
-  goldBalance: string | null
   exchangeRatePair: ExchangeRatePair | null
   fee: string
   appConnected: boolean
 }
 
-interface ConfirmationInput {
-  makerAmount: BigNumber
-  makerToken: Token
-}
-
 interface DispatchProps {
   fetchExchangeRate: typeof fetchExchangeRate
   exchangeTokens: typeof exchangeTokens
 }
 
+interface NavProps {
+  exchangeInput: {
+    makerToken: CURRENCY_ENUM
+    makerTokenBalance: string
+    inputToken: CURRENCY_ENUM
+    inputTokenCode: string
```

Comment:
Why do we need `inputTokenCode` given we already have `inputToken`?

Reply:
To avoid having to figure out the string again. In the future I would see us using CURRENCIES.code instead, but in the mean time I wanted to avoid copying over the `getInputTokenDisplayText` function